### PR TITLE
Add beat syncing container

### PR DESCRIPTION
**Pull request by smoogipoo**
Superseeds https://github.com/ppy/osu/pull/777.

### CODE REVIEW

**Comment by ColdVolcano** [comment]
Somehow the file is not in the csproj

**Comment by smoogipooo** [comment]
:PsyLikeThis:

File: osu.Game/Graphics/Containers/BeatSyncedContainer.cs [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+using osu.Game.Beatmaps.Timing;
+
+namespace osu.Game.Graphics.Containers
+{
+    public class BeatSyncedContainer : Container
+    {
+        /// <summary>
+        /// A new beat will not be sent if the time since the beat is larger than this tolerance.
+        /// </summary>
+        private const int seek_tolerance = 20;
```
Comment (peppy) [inline osu.Game/Graphics/Containers/BeatSyncedContainer.cs]: won't this container stop working completely at 50fps or less?

File: osu.Game/Graphics/Containers/BeatSyncedContainer.cs [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+        {
+            if (beatmap.Value?.Track == null)
+                return;
+
+            double currentTrackTime = beatmap.Value.Track.CurrentTime;
+            ControlPoint overridePoint;
+            ControlPoint controlPoint = beatmap.Value.Beatmap.TimingInfo.TimingPointAt(currentTrackTime, out overridePoint);
+
+            bool kiai = (overridePoint ?? controlPoint).KiaiMode;
+            int beat = controlPoint.BeatLength > 0 ? (int)((currentTrackTime - controlPoint.Time) / controlPoint.BeatLength) : 0;
```
Comment (peppy) [inline osu.Game/Graphics/Containers/BeatSyncedContainer.cs]: can't we just return early if `BeatLength == 0` and avoid this `?;`?

**Review by peppy — CHANGES_REQUESTED** [review]
as above

**Review by peppy — APPROVED** [review]

**Pull request merged**